.pre-commit-config.yaml

```diff
@@ -0,0 +1,15 @@
+# Pre-commit (https://pre-commit.com) configuration for assorted lint checks.
+#
+# See https://pre-commit.com/hooks.html for more hooks.
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+      - id: check-merge-conflict
+
+  - repo: https://github.com/psf/black
+    rev: 23.3.0
+    hooks:
+      - id: black
+        name: Run Black to format Python files
```

Comment:
There are some other hooks that we'll want but this is good as a starting point.

* https://github.com/iree-org/iree-test-suites/tree/main/linalg_ops has C++ source code that will be formatted with clang-format (https://github.com/iree-org/iree/blob/e8ee4801eda1babed0c513e941b88df3c3e25689/.pre-commit-config.yaml#L33-L41)
* Trailing whitespace, end of file fixer, and forbid tabs are generally useful
* check-yaml is useful for workflow files
* mypy type checking will be useful for python, but we can also have that run as part of the tests instead of lint